usr/share/rear/layout/prepare/GNU/Linux/130_include_filesystem_code.sh

```diff
@@ -183,38 +183,50 @@ function create_fs () {
         (btrfs)
             # Cleanup disk partition provided the disk partition is not already mounted:
             echo "mount | grep -q $device || $cleanup_command" >> "$LAYOUT_CODE"
+
             # Actually create the filesystem provided the disk partition is not already mounted.
-            # User -f [force] to force overwriting an existing btrfs on that disk partition
-            # when the disk was already used before, see https://bugzilla.novell.com/show_bug.cgi?id=878870
             (   echo "# if $device is already mounted, skip"
                 echo "# force overwriting existing btrfs when the disk was already used before"
-                echo "mount | grep -q $device || mkfs -t $fstype -f $device"
+                echo "if ! mount | grep -q $device >&2 ; then"
             ) >> "$LAYOUT_CODE"
+
+            if [ -n "$uuid" ] ; then
+                # Latest version of btrfs provides -U option to specify UUID druring the filesystem creation.
+                # User -f [force] to force overwriting an existing btrfs on that disk partition
+                # when the disk was already used before, see https://bugzilla.novell.com/show_bug.cgi?id=878870
+                (   echo "  # Try to create btrfs with UUID"
```

Comment:
Why not use a [here document](http://tldp.org/LDP/abs/html/here-docs.html), either `<<-EOF` to parse variables and indent with TABs or `<<'EOF'` to prevent variable replacements. I think that it will make it much more readable.

Another option is to define functions in the outer script and then add them. Also solves the quoting problem because you don't need to extra quote anything. Example:

```bash
function fun { echo "$HOME" ; } ; { declare -f fun ; echo fun ; } >/tmp/script.sh ; cat /tmp/script.sh
```